supervision/metrics/detection.py

```diff
@@ -456,3 +462,373 @@ def plot(
+            ...             [
+            ...                 [0.0, 0.0, 3.0, 3.0, 1],
+            ...                 [2.0, 2.0, 5.0, 5.0, 1],
+            ...                 [6.0, 1.0, 8.0, 3.0, 2],
+            ...             ]
+            ...         ),
+            ...         array([1.0, 1.0, 2.0, 2.0, 2]),
+            ...     ]
+            ... )
+
+            >>> predictions = [
+            ...     array(
+            ...         [
+            ...             [0.0, 0.0, 3.0, 3.0, 1, 0.9],
+            ...             [0.1, 0.1, 3.0, 3.0, 0, 0.9],
+            ...             [6.0, 1.0, 8.0, 3.0, 1, 0.8],
+            ...             [1.0, 6.0, 2.0, 7.0, 1, 0.8],
+            ...         ]
+            ...     ),
+            ...     array([[1.0, 1.0, 2.0, 2.0, 2, 0.8]])
+            ... ]
+
+            >>> mean_average_precison = sv.MeanAveragePrecision.from_tensors(
+            ...     predictions=predictions,
+            ...     targets=targets,
+            ... )
+
+            >>> mean_average_precison.map
+            0.2899
+            ```
+        """
+        cls._validate_input_tensors(predictions, targets)
+        map, map50, map75 = 0, 0, 0
+
+        class_index = 4
+        conf_index = 5
+
+        stats, average_precisions = [], []
+        iou_levels = np.linspace(0.5, 0.95, 10)
+        num_ious = iou_levels.size
+
+        for true_batch, detection_batch in zip(targets, predictions):
+            nl, npr = (
+                true_batch.shape[0],
+                detection_batch.shape[0],
+            )
+            correct = np.zeros((npr, num_ious), dtype=bool)  # init
+
+            if npr == 0:
+                if nl:
+                    stats.append((correct, *np.zeros((2, 0)), true_batch[:, 4]))
+                continue
+            if nl:
+                correct = cls._match_detection_batch(
+                    predictions=detection_batch,
+                    targets=true_batch,
+                    iou_levels=iou_levels,
+                )
+                stats.append(
+                    (
+                        correct,
+                        detection_batch[:, conf_index],
+                        detection_batch[:, class_index],
+                        true_batch[:, class_index],
+                    )
+                )
+
+        stats = [np.concatenate(x, 0) for x in zip(*stats)]
+
+        if len(stats) and stats[0].any():
+            average_precisions = cls.average_precisions_per_class(*stats)
+            ap50, ap75, average_precisions = (
+                average_precisions[:, 0],
+                average_precisions[:, 5],
+                average_precisions.mean(1),
+            )
+            map50, map75, map = ap50.mean(), ap75.mean(), average_precisions.mean()
+
+        return cls(
+            map=map, map50=map50, map75=map75, average_precisions=average_precisions
+        )
+
+    @staticmethod
+    def detections_to_tensor(detections: Detections) -> np.ndarray:
```

Comment:
@hardikdava, is there any reason you decided to create your implementation of `detections_to_tensor`? I'm not sure, but it looks like we could make `ConfusionMatrix.detections_to_tensor` a reusable utility. And use it in both places.

Reply:
Fixed. I had some issue with the results, so for debuging I created a new function. But I am now reusing it from Confusionmatrix.